webpack/module-minifier-plugin/src/GenerateLicenseFileForAsset.ts

```diff
@@ -22,15 +40,7 @@ export function generateLicenseFileForAsset(
 ): string {
   // Extracted comments from the minified asset and from the modules.
   // The former generally will be nonexistent (since it contains only the runtime), but the modules may have some.
-  const comments: Set<string> = new Set(asset.extractedComments);
-  for (const moduleId of asset.modules) {
-    const mod: IModuleInfo | undefined = minifiedModules.get(moduleId);
-    if (mod) {
-      for (const comment of mod.extractedComments) {
-        comments.add(comment);
-      }
-    }
-  }
+  const comments: Set<string> = new Set(iterateAllComments(asset.modules, minifiedModules));
```

Comment:
If you're just putting this into a set, why not just have the function return a set instead of using a generating function?